In [84]:
import pandas as pd 
import numpy as np

load dataset

In [85]:
df_independent = pd.read_csv('train_x.csv')

remove unwanted feature

In [86]:
df_independent = df_independent.drop(['Unnamed: 0'],1)

In [87]:
df_independent

shop_id  date_block_num  item_id  item_cnt_day
0              0               1    22154           1.0
1              0               0     2564           1.0
2              0               1     2572           1.0
3              0               1     2573           1.0
4              0               0     2574           1.0
...          ...             ...      ...           ...
1902212       59              33     5352           1.0
1902213       59              33     3831           1.0
1902214       59              33    15411           1.0
1902215       59              33    13637           1.0
1902216       59              33     7462           1.0

[1902217 rows x 4 columns]

find out quantities of unique labales of 'item_id','shop_id'

In [88]:
df_independent['item_id'].value_counts().sort_values()

0            1
1068         1
18245        1
21319        1
5966         1
         ...  
6675      2643
2808      2764
5822      3469
17717     7755
20949    11080
Name: item_id, Length: 21807, dtype: int64

In [89]:
import numpy as np

write a programm to arrange unique 'shop_id' with unique  'date_block_num' 
date_block_num: a consecutive month number, used for convenience. January 2013 is 0, February 2013 is 1,..., October 2015 is 33
shop_id - unique identifier of a shop

as example
shop_id  date_block_num  item_id  item_cnt_day
    1               1    22154           1.0
    
In February 2013 from shop 1, 1 product of  22154  'item_id' was  sold

In [8]:
df_k=[]
df_s=[]
for i in np.arange(0,59,1):
    # l=[i]
    df_k.append(df_independent.loc[lambda df_independent: df_independent['shop_id'] == i, :])
    df=pd.DataFrame(df_k[i])
    for j in np.arange(0,33,1):
         df_s.append(df.loc[lambda df:df['date_block_num'] == j, :])
            
       
        
 
        

In [9]:
df_s

[      shop_id  date_block_num  item_id  item_cnt_day
 1           0               0     2564           1.0
 4           0               0     2574           1.0
 6           0               0     2604           1.0
 7           0               0     2607           1.0
 9           0               0     2705           1.0
 ...       ...             ...      ...           ...
 5164        0               0    17934           1.0
 5165        0               0    16473           1.0
 5167        0               0    22039           1.0
 5168        0               0    21999           1.0
 5169        0               0    21831           1.0
 
 [2739 rows x 4 columns],
       shop_id  date_block_num  item_id  item_cnt_day
 0           0               1    22154           1.0
 2           0               1     2572           1.0
 3           0               1     2573           1.0
 5           0               1     2574           1.0
 8           0               1     2607           1.0


In [10]:
df_s[1]['shop_id'].value_counts().sort_values()

0    2826
Name: shop_id, dtype: int64

In [11]:
df_s[0]['shop_id'][:5]
     

1    0
4    0
6    0
7    0
9    0
Name: shop_id, dtype: int64

In [12]:
len(df_s)

1947

remove all blank dataframe

In [13]:
df_final=[]
for i in np.arange(0,1947,1):
    if df_s[i].size >0:
        df_final.append(df_s[i])  

In [14]:
len(df_final)

1509

In [15]:
for i in np.arange(0,1509,1):
    df_final[1]['item_id'].value_counts()

write aprogramm to determine total count of unique 'item_id' or items are being sold permonth from a unique shop 
as example
item_id  shop_id  item_cnt_month
   3432        2               4

total 4 nubers of items haveing 3432 "item_id" was being sold per month from 'shop_id'2        

In [36]:
df_pro=[]
df_c=[]

for i in np.arange(0,1509,1):
    df_c.append(df_final[i]['item_id'].value_counts())
    l1=list(df_c[i].index)
    l2=list(df_c[i])
    l3=df_final[i]['shop_id'][: len(df_c[i])]
    data = {'item_id': l1,
        'shop_id': l3,
           'item_cnt_month':l2}
    df_pro.append(pd.DataFrame(data))
    
    

remove 0 and 1 labaled shop id beacouse provided test sample dose cont cointain 0 and 1 labaled

In [37]:
df_for_train=list(df_pro[4:1509])

In [38]:
df_for_train

[       item_id  shop_id  item_cnt_month
 9152      3432        2               4
 9154      2808        2               4
 9165      3321        2               4
 9166      2920        2               4
 9174      2304        2               3
 ...        ...      ...             ...
 15340    15010        2               1
 15347     3328        2               1
 15356     3743        2               1
 15357    19083        2               1
 15359     9216        2               1
 
 [728 rows x 3 columns],
        item_id  shop_id  item_cnt_month
 9169      3328        2               3
 9175      2448        2               3
 9185       484        2               2
 9189      2750        2               2
 9227     19674        2               2
 ...        ...      ...             ...
 15892     8528        2               1
 15893    16210        2               1
 15894     9043        2               1
 15902     3318        2               1
 15914     7169        2      

In [26]:
len(df_for_train)

1505

prepare a funtion to randomely chose sample frome df_for_train and preapare training and testing sample 

In [74]:
def sample(j):
    df_j=[]
    for i in np.arange(0,1505,1):
          df_j.append(df_for_train[i].sample(n = j, replace = True)) 
    return df_j   

In [75]:
x=sample(10)

4O SAMPLE FROM EACH SET

In [76]:
train_x_1= pd.concat(sample(40),axis=0) 

In [77]:
train_x_2= pd.concat(sample(40),axis=0)

In [78]:
train_x_3= pd.concat(sample(40),axis=0)

In [79]:
train_x_4= pd.concat(sample(40),axis=0)

In [80]:
test_x_1= pd.concat(sample(40),axis=0)

8O SAMPLE FROM EACH SET FOR WORKING MODEL

In [81]:
TRAIN_TO_FIT=pd.concat(sample(80),axis=0)


In [82]:
train_x_1.to_csv('train_x_1.csv')
train_x_2.to_csv('train_x_2.csv')
train_x_3.to_csv('train_x_3.csv')
train_x_4.to_csv('train_x_4.csv')
test_x_1.to_csv('test_x_1.csv')


In [83]:
TRAIN_TO_FIT.to_csv('TRAIN_TO_FIT.csv')